engine/src/main/java/io/camunda/zeebe/engine/processing/dmn/EvaluateDecisionProcessor.java

```diff
@@ -91,19 +91,14 @@ private Either<Failure, PersistedDecision> getDecision(final DecisionEvaluationR
     final String decisionId = record.getDecisionId();
     final long decisionKey = record.getDecisionKey();
 
-    final var decisionIdProvided = !decisionId.isEmpty();
-    final var decisionKeyProvided = decisionKey != -1L;
-
-    if (decisionIdProvided == decisionKeyProvided) {
-      // XNOR if both ID and KEY are provided/missing
-      return Either.left(new Failure(ERROR_MESSAGE_NO_IDENTIFIER_SPECIFIED));
-    }
-
-    if (decisionIdProvided) {
+    if (!decisionId.isEmpty()) {
       return decisionBehavior.findDecisionById(decisionId);
       // TODO: expand DecisionState API to find decisions by ID AND VERSION (#11230)
-    } else {
+    } else if (decisionKey > -1L) {
       return decisionBehavior.findDecisionByKey(decisionKey);
+    } else {
+      // if both ID and KEY are missing
+      return Either.left(new Failure(ERROR_MESSAGE_NO_IDENTIFIER_SPECIFIED));
```

Comment:
🔧 It is a bit strange to me that we prioritise the decision id now. The key is more specific.

Reply:
It's the same for creating a process instance ([src](https://github.com/camunda/zeebe/blob/f549e5521664277517abb20f369f17edf47e2eb5/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java#L334-L345)). 

I suspect it's related to the issue I'm facing: In multiple languages, an empty string is the default value of a String, and this is an invalid ID value. However, the default value for Long is 0, which is a valid KEY value. And it's difficult to guarantee that the Zeebe client implementation for each language will override the default KEY value to -1. So it's easier to check the ID first, and the KEY second.